runtime/context.go

```diff
@@ -271,3 +271,36 @@ func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.
 func (c *contextLogger) Check(lvl zapcore.Level, msg string) *zapcore.CheckedEntry {
 	return c.log.Check(lvl, msg)
 }
+
+// ContextMetrics emits metrics
+type ContextMetrics struct {
+	scope                  tally.Scope
+	InboundHTTPMetrics     *InboundHTTPMetrics
+	EndpointMetrics        *EndpointMetrics
+	InboundTChannelMetrics *InboundTChannelMetrics
+}
+
+// NewContextMetrics emits metrics.
+func NewContextMetrics(scope tally.Scope) *ContextMetrics {
+	return &ContextMetrics{
+		scope: scope,
+	}
+}
+
+// MakeInboundHTTPMetrics add tags to scope and create inbound http metrics
+func (c *ContextMetrics) MakeInboundHTTPMetrics(tags map[string]string) {
+	scope := c.scope.Tagged(tags)
+	c.InboundHTTPMetrics = NewInboundHTTPMetrics(scope)
```

Comment:
This mutates the `InboundHTTPMetrics` field on a `ContextMetrics` object, which is part of an endpoint, not a request. And the function is called in `HandleRequest` https://github.com/uber/zanzibar/pull/488/files#diff-7aed2fb9afc6ff1780de1ca94cd0a938R146, which is called per request. It means the one `ContextMetrics` object is mutated by every request while the requests are concurrent. The race condition with this one is strong.

Reply:
fixed